docs/components.rst

```diff
@@ -2,9 +2,128 @@
+concept on the file system for now. This means that if a permission is applied to a group in `Magpie`_, `Cowbird`
+will detect the permission change but will not do anything, since the group on the file system does not correspond to
+the groups found on `Magpie`_. Also, it would not make sense to update the file associated to a resource for all the
+users of a group, since the file is supposed to be associated to a single user anyway.
+
+Note that even if group permission changes on `Magpie`_ are not handled by `Cowbird`, a group permission could still
+have an impact on permission resolution. For example, if a shapefile needs to allow a ``read`` permission on `Magpie`_,
+but that permission already resolves to ``allow`` because of a group permission, no permission will be added.
+
+.. _geoserver_file_layer_permissions:
+
+File/Layer permissions
+######################
+
+Also, file events will only be processed in the case of the ``.shp`` file, since it is considered to be the main
+component of a shapefile. The other extensions associated with a shapefile will not be processed if they trigger an
+event, and will only be updated in the case of a change on the ``.shp`` file.
+
+Note that in the case where a user has all the ``read`` permissions on `Magpie`_ for example, and a single one of
+them is deleted, `Cowbird` will not change the file permissions since other ``read`` permissions will still be found on
+`Magpie`_. This means that a synchronization will not be triggered and `Magpie`_ permissions will stay the same, meaning
+all the ``read`` permissions activated except for the one removed.
+If eventually a change is applied to the file (ex.: changing the permissions from ``r--`` to ``rw-``),
+it would trigger a synchronization, and the one `Magpie`_ permission that was removed earlier would be reenabled,
+because of the ``read`` permission found on the file.
+The same would apply if we use ``write`` permissions in this last example.
+
+Shapefiles will only be assigned ``read`` or ``write`` permissions on the file system. ``execute`` permissions are not
+needed for shapefiles.
+
+.. _geoserver_folder_workspace_permissions:
+
+Folder/Workspace permissions
+############################
+
+Workspaces will always keep their ``execute`` permissions even if they don't have any permissions enabled on `Magpie`_.
+This enables accessing the children files, in case the children resource has permissions enabled on `Magpie`_.
+Since a children resource has priority on `Magpie`_ if its permissions are enabled, it makes sense to allow the access
+to the file on the file system too. Note that if the folder only has ``execute`` permissions, the file will only be
+accessible via a direct path or url, and it will not be accessible via a file browser, or on the JupyterLab file
+browser. This should allow the user to still share his file using a path or url.
+
+.. _geoserver_operations_to_avoid:
+
+Operations to avoid
+###################
```

Comment:
These should probably be copied over into the README to appear in the JupyterHub workspace to warn users as best as possible.

Reply:
I added a note to the related Jira task so it can be added to the readme : [DAC-545](https://crim-ca.atlassian.net/browse/DAC-545)